docs/pages/deploy-a-cluster/teleport-enterprise/gcp-kms.mdx

```diff
@@ -0,0 +1,270 @@
+description: 'Teleport permissions for using KMS keys'
+stage: ALPHA
+includedPermissions:
+- cloudkms.cryptoKeys.create
+- cloudkms.cryptoKeys.list
+- cloudkms.cryptoKeyVersions.create
+- cloudkms.cryptoKeyVersions.destroy
+- cloudkms.cryptoKeyVersions.useToSign
+- cloudkms.cryptoKeyVersions.viewPublicKey
+```
+
+```code
+$ gcloud iam roles create teleport_kms_role \
+    --project <Var name="GCP-Project-ID" description="Your GCP project ID"/> \
+    --file teleport_kms_role.yaml \
+    --format yaml
+```
+
+Note the `name` field in the output which is the fully qualified name for the
+custom role and must be used in later steps.
+
+```code
+$ export IAM_ROLE=<role name output from above>
+```
+
+If you don't already have a GCP service account for your Teleport Auth Server
+you can create one with the following command, otherwise use your existing
+service account.
+
+```code
+$ gcloud iam service-accounts create teleport-auth-server \
+    --description="Service account for Teleport Auth Server" \
+    --display-name="Teleport Auth Server" \
+    --format=yaml
+```
+
+Note the `email` field in the output, this must be used as the identifier for
+the service account.
+
+```code
+$ export SERVICE_ACCOUNT=<email output from above command>
+```
+
+Create the IAM policy binding to grant the role to the service account for this
+keyring.
+
+```code
+$ gcloud kms keyrings add-iam-policy-binding "teleport-keyring" \
+    --location "global" \
+    --member "serviceAccount:${SERVICE_ACCOUNT}" \
+    --role "${IAM_ROLE}"
+```
+
+## Step 3/5. Provide the service account credentials to your Auth Server
+
+The Teleport Auth Server will use Application Default Credentials to make
+requests to the GCP KMS service.
+Provide credentials for the `teleport-auth-server` service account created in
+step 2 to the Application Default Credentials of the environment you are running
+your Teleport Auth Server in.
+Supported environments include GCE VMs, GKE pods, and others.
+
+See the GCP docs for
+[Application Default Credentials](https://cloud.google.com/docs/authentication/provide-credentials-adc)
+to learn how to provide them for your preferred environment.
+
+To make sure the credentials have been configured correctly, you can run the
+`gcloud` CLI tool from your Teleport Auth Server's environment. Some example
+commands you could use to debug are listed here:
+
+```code
+$ export KEYRING="<Var name="teleport-keyring"/>"
+$ gcloud kms keys list --location global --keyring "${KEYRING}"
+$ gcloud kms keys create --keyring "${KEYRING}" --location global "test-key" \
+    --purpose asymmetric-signing \
+    --default-algorithm rsa-sign-pkcs1-4096-sha512
+$ echo hello > /tmp/hello.txt
+$ gcloud kms asymmetric-sign --keyring "${KEYRING}" --location global \
+    --key "test-key" --version 1 \
+    --input-file /tmp/hello.txt --signature-file /tmp/hello.sig
+$ gcloud kms keys versions destroy --keyring "${KEYRING}" --location global --key "test-key" 1
```

Comment:
I think it would be an improvement, if possible, to include some form of example output for these commands, to show what form of response indicates a positive response.